src/main/java/org/sonatype/repository/vgo/internal/hosted/VgoHostedFacetImpl.java

```diff
@@ -120,13 +122,20 @@ public Content getList(final String module) {
 
     StorageTx tx = UnitOfWork.currentTx();
 
-    Iterable<Asset> assetsForModule = vgoDataAccess.findPackageAssetsForModule(tx, getRepository(), module);
+    Iterable<Asset> assetsForModule = vgoDataAccess.findAssetsForModule(tx, getRepository(), module);
 
-    String listOfVersions = StreamSupport.stream(assetsForModule.spliterator(), false)
-        .map(asset -> asset.name())
+    List<String> collection = StreamSupport.stream(assetsForModule.spliterator(), false)
+        .filter(asset -> VGO_PACKAGE.name().equals(asset.formatAttributes().get(P_ASSET_KIND)))
```

Comment:
Question: why the `filter` vs the `where clause`?

Reply:
Searching on attributes map can be extremely slow because it can cause a full table scan. Filtering out in software is less expensive.